chrome/content/zotero/xpcom/sync/syncAPIClient.js

```diff
@@ -430,6 +434,46 @@ Zotero.Sync.APIClient.prototype = {
 	}),
 	
 	
+	getAPIKeyFromCredentials: Zotero.Promise.coroutine(function* () {
+		var apiKey;
+		let body = JSON.stringify({
+			username: this.username,
+			password: this.password,
+			name: Zotero.isStandalone ? "Zotero Standalone" : "Zotero for Firefox",
```

Comment:
"Automatic Zotero Client Key" — user agent will be determined automatically, and we might share prefs between Firefox and Standalone at some point